### PR TITLE
Live updates for materials

### PR DESCRIPTION
**Pull request by neph1**
I managed to break out this as well. Still one annoying "whitespace" diff in this one.
I made a preliminary merge test with these PR's into the big one and it was fairly smooth, so I'm less worried about that now.

### CODE REVIEW

File: jme3-materialeditor/src/com/jme3/gde/materialdefinition/EditableMatDefFile.java [hunk 1]
```diff
@@ -1,408 +1,400 @@
...
-            //material.selectTechnique("Default", SceneApplication.getApplication().getRenderManager());
-            if (matToRemove != null) {
-                for (MatParam matParam : matToRemove.getParams()) {
-                    try {
-                        material.setParam(matParam.getName(), matParam.getVarType(), matParam.getValue());
-                    } catch (IllegalArgumentException ie) {
-                        matToRemove.clearParam(matParam.getName());
-                    }
-
-                }
```
Comment (neph1) [inline jme3-materialeditor/src/com/jme3/gde/materialdefinition/EditableMatDefFile.java:264]: This is what caused the mat previews not to update when changing parameters. They got overwritten when the material was reloaded.

File: jme3-materialeditor/src/com/jme3/gde/materialdefinition/editor/ShaderNodeDiagram.java [hunk 2]
```diff
@@ -1,492 +1,539 @@
...
+        @Override
+        public void run() {
+            while(running) {
+                if (backDrop.isVisible() && !backDrop.getRenderer().isPreviewRequested()) {
+                    backDrop.refreshOnly();
+                }
+                try {
+                    Thread.sleep(20);
+                } catch (InterruptedException ex) {
+                    running = false;
```
Comment (MeFisto94) [inline jme3-materialeditor/src/com/jme3/gde/materialdefinition/editor/ShaderNodeDiagram.java:520]: Always call Thread.currentThread.interupt() in an InterruptedException

**Review by MeFisto94 — APPROVED** [review]
In general looking very good, just a few minor considerations

**Comment by neph1** [comment]
Yeah, those comments are from before. May I merge these once I've done the fixes? I would like to merge one PR at a time and sync them up to my main branch before doing the next.

File: jme3-materialeditor/src/com/jme3/gde/materialdefinition/editor/ShaderNodeDiagram.java [hunk 3]
```diff
@@ -1,492 +1,539 @@
...
+        if (SwingUtilities.isLeftMouseButton(e)) {
+            if (draggedFrom == null) {
+                for (Selectable selectedItem : selectedItems) {
+                    if (selectedItem instanceof ShaderOutBusPanel) {
+                        ShaderOutBusPanel bus = (ShaderOutBusPanel) selectedItem;
+                        MouseEvent me = SwingUtilities.convertMouseEvent(this, e, bus);
+                        bus.dispatchEvent(me);
+                    }
+                }
+//                if (!e.isConsumed()){
```
[the remark this answers is not in the document]
Comment (neph1) [reply jme3-materialeditor/src/com/jme3/gde/materialdefinition/editor/ShaderNodeDiagram.java:136]: Haha, yeah. I've removed that from several branches, but not this one, apparently. I think it was part of one of the first things I tried.

**Pull request merged**